package.json

```diff
@@ -2,23 +2,31 @@
   "name": "webpack-filter-warnings-plugin",
   "version": "1.2.1",
   "description": "Allows you to hide certain warnings from webpack compilations",
-  "main": "dist/cjs.js",
+  "main": "cjs/index.js",
+  "types": "cjs/index.d.ts",
   "scripts": {
-    "test": "jest",
-    "start": "npm run build -- -w",
-    "build": "cross-env NODE_ENV=production babel src -d dist --ignore 'src/**/*.test.js'",
-    "clean": "del-cli dist",
-    "lint": "eslint --cache src test",
-    "lint-staged": "lint-staged",
     "prebuild": "npm run clean",
+    "build": "npm run build:typescript && npm run build:babel",
+    "build:babel": "babel ts-out --out-dir . --copy-files ",
+    "build:cjs": "tsc -p './build-tools/tsconfig-cjs.json'",
+    "build:dev": "npm run build -- -- --source-maps inline",
+    "build:es": "tsc -p './build-tools/tsconfig-es.json'",
```

Comment:
I think this is unneeded now as we're only shipping the commonjs to npm?

Reply:
This still builds different code part that is accessible in the package via `webpack-filter-warnings-plugin/es`. CJS build generated by other pipeline is transpiled to be compatible with Node 4.x - so e.g. classes, spread operators etc. are replaced with compatible code, so it's essentially slower on newer NodeJS versions than `es` build.

Reply:
There aren't really any benefits to shipping an ES module version as well (today at least), as node still doesn't have native support for ES modules, so no one would be able to import this version without using a transpiler. Regarding the performance implications, it's such a small module that I doubt there would be any noticeable benefits to the end user. In fact, as node 6 reaches EOL next month, I think we should just drop support for it all together and require >= node 8 to use this package. As we're doing to bump the major anyway, I think that's acceptable. WDYT?

Reply:
Yeah, I'm a bit of performance freak, so I got carried away :)

I completely agree, especially as babelified files contain flag that allows ES-compatible code to reuse module (`__esModule: true`), so it may be used without any problems in e.g. Typescript and other environments.

What if I adjust `.babelrc` and `package.json` so it targets at least Node 8 and drop `ES` config?

Reply:
Yup that sounds great! Can you also adjust the [node engines property](https://github.com/mattlewis92/webpack-filter-warnings-plugin/blob/master/package.json#L40) to be `>= 8.0.0`